File: dotnet/Multiplayer/Rooms/Actions/ActionCompleteRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Rooms.Actions
{
    /// <summary>
    /// Request data for completing an action in a room.
    /// Serializes completion status and response data for API transmission.
    /// </summary>
    /// <typeparam name="T">The type of response data.</typeparam>
    public class ActionCompleteRequest<T> where T : class, new()
    {
        [JsonInclude]
        private string Status { get; set; } = ERoomCompleteActionStatus.Completed.ToString().ToLower();
        [JsonInclude]
        private T? Response_data { get; set; }

        /// <summary>
        /// Initializes a new ActionCompleteRequest.
        /// </summary>
        /// <param name="status">The status of the action completion.</param>
        /// <param name="responseData">The response data to return with the completion.</param>
        public ActionCompleteRequest(ERoomCompleteActionStatus status, T? responseData)
        {
            this.Status = status.ToString().ToLower();
            this.Response_data = responseData;
        }
    }
}

File: java/src/main/java/com/michitai/multiplayer/rooms/actions/ActionCompleteRequest.java
package com.michitai.multiplayer.rooms.actions;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Request for completing an action with a response.
 */
public class ActionCompleteRequest {
    @JsonProperty("status")
    private String status;

    @JsonProperty("response_data")
    private String responseData;

    public ActionCompleteRequest(String status, String responseData) {
        this.status = status;
        this.responseData = responseData;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getResponseData() {
        return responseData;
    }

    public void setResponseData(String responseData) {
        this.responseData = responseData;
    }
}
